File: core/feature48.js
function feature48() {
  var parameters = extractParameters(["D", "A", "B", "L", "V", "T"]);
  return feature48_core(parameters);
}

function feature48_core(parameters) {
  if (parameters == null) {
    return null;
  }
  var result = {};
  var D = getParameter(parameters, "D")
  var A = getParameter(parameters, "A")
  var B = getParameter(parameters, "B")
  var L = getParameter(parameters, "L")
  var V = getParameter(parameters, "V")
  var T = getParameter(parameters, "T")
  var Rad = Math.PI / 180.0
  var r = D / 2.0
  var M = A / 2.0
  var N = B / 2.0
  var I = Math.asin(V / L) / Rad
  result["S"] = 2.0 * Math.PI * r / T
  result["F1"] = Math.sqrt(Math.pow(M - V - r * Math.cos(I * Rad), 2.0) + Math.pow(L * Math.cos(I * Rad) - r * Math.sin(I * Rad), 2.0))
  var pa = [];
  for (var X = 0.0; X <= T / 4.0; X += 1.0) {
    var F = X * (360.0 / T)
    var Q = Math.sqrt(Math.pow(M - (V + r * Math.cos(I * Rad) * Math.cos(F * Rad)), 2.0) + Math.pow(N - r * Math.sin(F * Rad), 2.0) + Math.pow(L * Math.cos(I * Rad) - r * Math.sin(I * Rad) * Math.cos(F * Rad), 2.0))
    pa.push(Q)
  }
  result["PointsA"] = pa
  var pb = [];
  for (var X = 0.0; X <= T / 4.0; X += 1.0) {
    var F = X * (360.0 / T)
    var Q = Math.sqrt(Math.pow(M + V + r * Math.cos(I * Rad) * Math.cos((90.0 + F) * Rad), 2.0) + Math.pow(N - r * Math.sin((90.0 + F) * Rad), 2.0) + Math.pow(L * Math.cos(I * Rad) - r * Math.sin(I * Rad) * Math.cos((90.0 + F) * Rad), 2.0))
    pb.push(Q)
  }
  result["PointsB"] = pb
  result["F2"] = Math.sqrt(Math.pow(M + V - r * Math.cos(I * Rad), 2.0) + Math.pow(L * Math.cos(I * Rad) + r * Math.sin(I * Rad), 2.0))
  var pm = [];
  for (var X = 0.0; X <= T / 4.0; X += 1.0) {
    var F = X * (360.0 / T)
    var Q = (A / (A + 2.0 * (V + r * Math.cos(I * Rad) * Math.cos((90.0 + F) * Rad)))) * Math.sqrt(Math.pow(M + V + r * Math.cos(I * Rad) * Math.cos((90.0 + F) * Rad), 2.0) + Math.pow(N - r * Math.sin((90.0 + F) * Rad), 2.0) + Math.pow(L * Math.cos(I * Rad) - r * Math.sin(I * Rad) * Math.cos((90.0 + F) * Rad), 2.0))
    pm.push(Q)
  }
  result["PointsM"] = pm
  return result;
}

File: core/feature48.go
package main

import "math"

func feature48(arguments map[string]interface{}) map[string]interface{} {
	parameters := extractParameters(arguments, []string{"D", "A", "B", "L", "V", "T"})
	if parameters == nil {
		return nil
	}
	result := make(map[string]interface{})
	D := getParameter(parameters, "D")
	A := getParameter(parameters, "A")
	B := getParameter(parameters, "B")
	L := getParameter(parameters, "L")
	V := getParameter(parameters, "V")
	T := getParameter(parameters, "T")
	Rad := math.Pi / 180.0
	r := D / 2.0
	M := A / 2.0
	N := B / 2.0
	I := math.Asin(V/L) / Rad
	result["S"] = 2.0 * math.Pi * r / T
	result["F1"] = math.Sqrt(math.Pow(M-V-r*math.Cos(I*Rad), 2.0) + math.Pow(L*math.Cos(I*Rad)-r*math.Sin(I*Rad), 2.0))
	pa := make([]interface{}, 0)
	for X := 0.0; X <= T/4.0; X += 1.0 {
		F := X * (360.0 / T)
		Q := math.Sqrt(math.Pow(M-(V+r*math.Cos(I*Rad)*math.Cos(F*Rad)), 2.0) + math.Pow(N-r*math.Sin(F*Rad), 2.0) + math.Pow(L*math.Cos(I*Rad)-r*math.Sin(I*Rad)*math.Cos(F*Rad), 2.0))
		pa = append(pa, Q)
	}
	result["PointsA"] = pa
	pb := make([]interface{}, 0)
	for X := 0.0; X <= T/4.0; X += 1.0 {
		F := X * (360.0 / T)
		Q := math.Sqrt(math.Pow(M+V+r*math.Cos(I*Rad)*math.Cos((90.0+F)*Rad), 2.0) + math.Pow(N-r*math.Sin((90.0+F)*Rad), 2.0) + math.Pow(L*math.Cos(I*Rad)-r*math.Sin(I*Rad)*math.Cos((90.0+F)*Rad), 2.0))
		pb = append(pb, Q)
	}
	result["PointsB"] = pb
	result["F2"] = math.Sqrt(math.Pow(M+V-r*math.Cos(I*Rad), 2.0) + math.Pow(L*math.Cos(I*Rad)+r*math.Sin(I*Rad), 2.0))
	pm := make([]interface{}, 0)
	for X := 0.0; X <= T/4.0; X += 1.0 {
		F := X * (360.0 / T)
		Q := (A / (A + 2.0*(V+r*math.Cos(I*Rad)*math.Cos((90.0+F)*Rad)))) * math.Sqrt(math.Pow(M+V+r*math.Cos(I*Rad)*math.Cos((90.0+F)*Rad), 2.0)+math.Pow(N-r*math.Sin((90.0+F)*Rad), 2.0)+math.Pow(L*math.Cos(I*Rad)-r*math.Sin(I*Rad)*math.Cos((90.0+F)*Rad), 2.0))
		pm = append(pm, Q)
	}
	result["PointsM"] = pm
	return result
}
